### ISSUE TITLE
grnn_shapenet_train and grnn_shapenet_eval not working properly

### ISSUE THREAD

**Issue by st2yang** [ricsonc/grnn#3]
Hi Ricson,

I try to train and test GRNNs on shapenet to reproduce the results in the paper. But there are some strange observations

1. I run the command python main.py grnn_shapenet_train. After around 3k iterations, the recon-loss become **nan**
2. I noticed the opname in option groups is **gqn3d**. Shoud not it be **reconstruction**?
3. I tried to run python main.py grnn_shapenet_eval after putting and renaming pretrained models properly. But the error says "raise Exception("checkpoint not found? (1)")".
FYI, the path is path_to_repo/ckpt/grnn_shapenet_train/main_weights (I also tried grnn_shapenet_trained, still wrong)

I tried to skim options.py and related files, and I think opname is modified to gqn3d even the training flag is grnn_shapenet_train. I am a little confused how to reproduce the results at the moment. Can you help to clarify a little bit?

Thank you in advance!
Yang

**Comment by ricsonc** [comment]
1. hm i've not experienced that before. how was the training going before the 3k iterations? 
2. no, gqn3d is correct. gqn3d just refers to our model with its 3D feature memory.
3. this is due to a mistake we made while cleaning up the code, which breaks our pretrained checkpoints. if you checkout the "v2" branch of this repository, those checkpoints *should* work -- let me know if you still have problems on that branch

the "v2" branch might also fix your training problems, although maybe not. if not, also let me know if you're using a different version of tensorflow as we are, and if you made any other tweaks to the code

thanks!

**Comment by st2yang** [comment]
Hi Ricson,

I checked the package and found some inconsistent versions (including tensorflow). After fixing these, now the training is smooth. Also the pretrained models work in v2 branch. Thanks a lot for your help!

I am planning to adopt GRNNs on real robot, by letting the robot arm automatically collect images, and use it for down-stream robotic applications. Any idea or suggestion about this? Such as difficulty or if GRNNs is applicable in real domain.

Thank you,
Yang

**Comment by ricsonc** [comment]
You may be interested in this ICLR submission which builds upon this work in the robotics domain. https://openreview.net/forum?id=Bklfcxrtvr 

It has also been tested on real scenes in this work. https://openreview.net/pdf?id=BJxt60VtPr

Good luck!
